apps/src/lib/client/utils.rs

```diff
@@ -1068,3 +1106,59 @@ pub fn validator_pre_genesis_file(pre_genesis_path: &Path) -> PathBuf {
 pub fn validator_pre_genesis_dir(base_dir: &Path, alias: &str) -> PathBuf {
     base_dir.join(PRE_GENESIS_DIR).join(alias)
 }
+
+// Exits the process if a directory already exists for a given chain.
+fn exit_if_chain_dir_exists(base_dir: &Path, chain_id: &ChainId) {
+    if fs::canonicalize(base_dir.join(chain_id.as_str())).is_ok() {
+        eprintln!("The chain directory for {} already exists.", chain_id);
+        cli::safe_exit(1);
+    }
+}
+
+// Decode and unpack the bytes of a release archive (.tar.gz) into `unpack_dir`.
+// Panics if this is not possible.
+fn unpack_release<R: std::io::Read>(tgz: R, unpack_dir: &Path) {
+    let decoder = GzDecoder::new(tgz);
+    let mut archive = tar::Archive::new(decoder);
+    archive.unpack(&unpack_dir).unwrap()
+}
+
+/// Returns the [`ChainId`] of the chain for a release archive unpacked at
+/// `unpacked_dir`.
+fn extract_chain_id(release_dir: &Path) -> eyre::Result<ChainId> {
+    let global_config_path = release_dir
+        .join(config::DEFAULT_BASE_DIR)
+        .join(config::global::FILENAME);
+    let global_config_str = std::fs::read_to_string(global_config_path)?;
+    let global_config: GlobalConfig = toml::from_str(&global_config_str)?;
+    Ok(global_config.default_chain_id)
+}
```

Comment:
Maybe there is a better way to work out the chain ID of a release archive than looking at the `global-config.toml` inside it. This is also assuming that a release archive will contain exactly one chain (I think that is the case?)